src/primitive/primitive2.rs

```diff
@@ -0,0 +1,129 @@
+//! Wrapper enum for 2D primitives
+
+use cgmath::{BaseFloat, Point2, Vector2};
+use cgmath::prelude::*;
+
+use {Aabb2, Ray2};
+use prelude::*;
+use primitive::{Circle, ConvexPolygon, Particle2, Rectangle};
+
+/// Wrapper enum for 2D primitives, that also implements the `Primitive` trait, making it easier
+/// to use many different primitives in algorithms.
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "eders", derive(Serialize, Deserialize))]
+pub enum Primitive2<S> {
+    /// Particle
+    Particle(Particle2<S>),
+    /// Circle
+    Circle(Circle<S>),
+    /// Rectangle
+    Rectangle(Rectangle<S>),
+    /// Convex polygon with any number of vertices.
+    ConvexPolygon(ConvexPolygon<S>),
+}
+
+impl<S> Into<Primitive2<S>> for Particle2<S> {
```

Comment:
should rather implement `From<Particle2<S>> for Primitive2<S>` and so on, since the `Into` would be automatic in this case

Reply:
For some reason I thought it was the other way around, silly me.